tempodb/backend/s3/s3_test.go

```diff
@@ -118,17 +121,17 @@ func TestReadError(t *testing.T) {
 	assert.Equal(t, wups, errB)
 }
 
-func fakeServerWithTags(t *testing.T, obj *url.Values) *httptest.Server {
+func fakeServerWithHeader(testedHeaderName string, t *testing.T, obj *url.Values) *httptest.Server {
```

Comment:
nit:  It's convention to keep `t *testing.T` as the first argument, can we swap this around?